mu-plugins/central-hub/tests/phpunit/unit/meta-data/remapCustomFieldsConfig.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Tests for the function remap_custom_fields_config().
+ *
+ * @package     spiralWebDb\centralHub\Tests\Unit\Metadata
+ * @since       1.3.0
+ * @author      Robert Gadon <rgadon107>
+ * @link        https://github.com/rgadon107/cornerstone
+ * @license     GNU-2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Unit\Metadata;
+
+use Brain\Monkey;
+use function spiralWebDB\Metadata\remap_custom_fields_config;
+use spiralWebDb\Cornerstone\Tests\Unit\Test_Case;
+
+/**
+ * Class Tests_RemapCustomFieldsConfig
+ *
+ * @package spiralWebDb\centralHub\Tests\Unit\Metadata
+ * @group   meta-data
+ */
+class Tests_RemapCustomFieldsConfig extends Test_Case {
+
+	/**
+	 * Prepare the test environment before each test.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once CENTRAL_HUB_ROOT_DIR . '/src/meta-data/meta-box.php';
+	}
+
+	/**
+	 * Clean up the test environment after each test.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+		$config = [];
+	}
+
+	/**
+	 *  Test remap_custom_fields_config() should return the default remapped config array when the given config is an
+	 *  empty array.
+	 */
+	public function test_should_return_default_remapped_config_array_when_given_config_is_empty_array() {
+		$config          = [
+			'meta_box.events' => [
+				'custom_fields' => [],
+			],
+		];
+		$remapped_config = [
+			'is_single'    => [],
+			'default'      => [],
+			'delete_state' => [],
+			'sanitize'     => [],
+		];
+		Monkey\Functions\expect( 'spiralWebDB\Metadata\remap_custom_fields_config' )
+			->once()
+			->with( $config )
+			->andReturn( $remapped_config );
+
+		$this->assertSame( $remapped_config, remap_custom_fields_config( $config ) );
+	}
+
+	/**
+	 * Test remap_custom_fields_config() should remap a config structure when the given config has post meta data.
+	 */
+	public function test_should_remap_config_structure_when_given_config_has_post_meta_data() {
+		$config          = [
+			'meta_box.events' => [
+				'custom_fields' => [
+					'event-date' => '09-24-2019',
```

Comment:
@rgadon107 These custom field configurations are not correct. Why? [Take a look at the Event's config](https://github.com/rgadon107/cornerstone/blob/add/remapCustomFieldsConfig/plugins/events/config/events.php#L43). Notice how the `'event-date'` field is configured:

https://github.com/rgadon107/cornerstone/blob/656b147274c10aa4dac6f3db269f2ac35643359a/plugins/events/config/events.php#L45-L57

You are _telling_ the Meta Data Module in the Central Hub how to set up and understand each custom field:

* Is it a single piece of meta data or an array?
* What's the default value?
* What's the value in the `$_POST` when it should be deleted?
* How should the value be sanitized?

Notice that each of these parameters aligns to the remapped configuration array.

Reply:
Custom fields `$config` array restructured for second unit test (passed).